### ISSUE TITLE
Error installing on Ruby v2.2

### ISSUE THREAD

**Issue by viniciusnz** [taf2/curb#234]
Hello,

I was trying to install curb on ruby v2.2 and got
```
make "DESTDIR="
compiling curb_upload.c
compiling curb_multi.c
curb_multi.c: In function ‘curl_multi_free’:
curb_multi.c:72:3: warning: implicit declaration of function ‘RHASH’ [-Wimplicit-function-declaration]
curb_multi.c:72:79: error: invalid type argument of ‘->’ (have ‘int’)
curb_multi.c: In function ‘ruby_curl_multi_remove’:
curb_multi.c:270:19: warning: variable ‘rbce’ set but not used [-Wunused-but-set-variable]
make: *** [curb_multi.o] Error 1
```

I tried updating my local
```
sudo apt-get install libcurl4-openssl-dev
```
as a first guess but it didn't work. I guess I'll stick with ruby v2.1.2p95 for now, but just wanted to let you know...

All the best!

**Comment by simo2409** [comment]
I have a similar problem on ruby 2.2 on CentOS 7:
````
make "DESTDIR="
compiling curb_errors.c
compiling curb_multi.c
curb_multi.c: In function ‘curl_multi_free’:
curb_multi.c:67:3: warning: implicit declaration of function ‘RHASH’ [-Wimplicit-function-declaration]
   if (rbcm && !rbcm->requests == Qnil && rb_type(rbcm->requests) == T_HASH && RHASH_LEN(rbcm->requests) > 0) {
   ^
In file included from curb_easy.h:10:0,
                 from curb_multi.c:14:
curb.h:44:38: error: invalid type argument of ‘->’ (have ‘int’)
   #define RHASH_LEN(hash) RHASH(hash)->tbl->num_entries
                                      ^
curb_multi.c:67:79: note: in expansion of macro ‘RHASH_LEN’
   if (rbcm && !rbcm->requests == Qnil && rb_type(rbcm->requests) == T_HASH && RHASH_LEN(rbcm->requests) > 0) {
                                                                               ^
curb_multi.c: In function ‘ruby_curl_multi_remove’:
curb_multi.c:265:19: warning: variable ‘rbce’ set but not used [-Wunused-but-set-variable]
   ruby_curl_easy *rbce;
                   ^
curb_multi.c: In function ‘ruby_curl_multi_perform’:
curb_multi.c:581:7: warning: implicit declaration of function ‘rb_thread_select’ [-Wimplicit-function-declaration]
       rc = rb_thread_select(maxfd+1, &fdread, &fdwrite, &fdexcep, &tv);
       ^
make: *** [curb_multi.o] Errore 1

make failed, exit code 2
````

I have already installed libcurl.x86_64 and libcurl-devel.x86_64 (both 7.29.0-19.el7)

**Comment by bsodmike** [comment]
I'm seeing something similar in Mac OS X 10.10.1

```
compiling curb_multi.c
curb_multi.c:69:79: warning: implicit declaration of function 'RHASH' is invalid in C99 [-Wimplicit-function-declaration]
  if (rbcm && !rbcm->requests == Qnil && rb_type(rbcm->requests) == T_HASH && RHASH_LEN(rbcm->requests) > 0) {
                                                                              ^
./curb.h:44:27: note: expanded from macro 'RHASH_LEN'
  #define RHASH_LEN(hash) RHASH(hash)->tbl->num_entries
                          ^
curb_multi.c:69:79: error: member reference type 'int' is not a pointer
  if (rbcm && !rbcm->requests == Qnil && rb_type(rbcm->requests) == T_HASH && RHASH_LEN(rbcm->requests) > 0) {
                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
./curb.h:44:40: note: expanded from macro 'RHASH_LEN'
  #define RHASH_LEN(hash) RHASH(hash)->tbl->num_entries
                          ~~~~~~~~~~~  ^
curb_multi.c:514:12: warning: implicit declaration of function 'rb_thread_select' is invalid in C99 [-Wimplicit-function-declaration]
      rc = rb_thread_select(maxfd+1, &fdread, &fdwrite, &fdexcep, &tv);
           ^
2 warnings and 1 error generated.
make: *** [curb_multi.o] Error 1

make failed, exit code 2
```

**Comment by taf2** [comment]
Please try master.

**Comment by bsodmike** [comment]
Thanks @taf2 that works :+1:

**Comment by taf2** [comment]
great - i'll try to push out a new gem later today then.

**Comment by paulspringett** [comment]
@taf2 did a new gem version get released for this please?

**Comment by bsodmike** [comment]
I just patched this in another project as well, grabbing master for now.

**Comment by davidw** [comment]
I'm having this problem with Ruby 2.2.1 on Ubuntu 13.04.  Moving to 'master' fixes it.

**Comment by lfender6445** [comment]
for those wondering, appears resolved in versions of curb `'~> 0.8.4'`
